builtin/mainmenu/dlg_create_world.lua

```diff
@@ -90,12 +90,16 @@ local function create_world_formspec(dialogdata)
 		-- should never happen but just pick the first game
 		game = pkgmgr.games[1]
 		core.settings:set("menu_last_game", game.id)
+
+		current_mg = dialogdata.mg or core.settings:get("mg_name")
```

Comment:
redundant, as the below code will run anyway, no?

Reply:
the if only runs if the game indicated by "menu_last_game" does not exist. maybe you misplaced the line.